### PR TITLE
[SPARK-10380] Confusing examples in pyspark SQL docs

### PR DESCRIPTION
**Pull request by GayathriMurali**
#8647

JIRA : https://issues.apache.org/jira/browse/SPARK-10380

Fixed confusing pyspark examples in SQL docs

### CODE REVIEW

File: python/pyspark/sql/dataframe.py [hunk 1]
```diff
@@ -974,6 +974,35 @@ def dropDuplicates(self, subset=None):
...
+        | 10|    80|Alice|
+        +---+------+-----+
+
+        >>> df.drop_duplicates(['name', 'height']).show()
+        +---+------+-----+
+        |age|height| name|
+        +---+------+-----+
+        |  5|    80|Alice|
+        +---+------+-----+
+        """
```
Comment (marmbrus) [inline python/pyspark/sql/dataframe.py]: @rxin and I are worried about these docs diverging overtime.  I think we should have it delegate (as it does today) and have the docs simply say ":func:`dropDuplicates` is an alias for :func:`drop_duplicates`."

Comment (GayathriMurali) [inline python/pyspark/sql/dataframe.py]: Would you like to keep either one of them in the doc and just include the note that say "x is an alias for y"? Should I do the same for astype/cast as well?

Comment (rxin) [inline python/pyspark/sql/dataframe.py]: I think Michael was referring to keeping only one sentence in the doc.

Comment (cloud-fan) [inline python/pyspark/sql/dataframe.py]: I think we can simply do `drop_duplicates  = dropDuplicates `, one example is https://github.com/apache/spark/blob/v1.6.0/python/pyspark/sql/dataframe.py#L904

Comment (marmbrus) [inline python/pyspark/sql/dataframe.py]: If you just do equals like that then in the docs the examples are not calling the function that is being defined, which a user complained was confusing.

Comment (GayathriMurali) [inline python/pyspark/sql/dataframe.py]: Should we remove the example and just have one sentence which says "drop_duplicates() alias for dropDuplicates()

Comment (marmbrus) [inline python/pyspark/sql/dataframe.py]: Yes

**Comment by GayathriMurali** [comment]
@marmbrus @rxin Let me know if there is anything else I need to do, to help merge this patch

**Comment by marmbrus** [comment]
ok to test

File: python/pyspark/sql/dataframe.py [hunk 2]
```diff
@@ -974,6 +974,11 @@ def dropDuplicates(self, subset=None):
             jdf = self._jdf.dropDuplicates(self._jseq(subset))
         return DataFrame(jdf, self.sql_ctx)
 
+    def dropDuplicates(self, subset=None):
+        """
+        :func:`drop_duplicates` is an alias for :func:`dropDuplicates`.
+        """
```
Comment (marmbrus) [inline python/pyspark/sql/dataframe.py]: Don't you need to actually delegate here?  Right now this looks like a no-op.

**Comment by rxin** [comment]
I subimtted https://github.com/apache/spark/pull/11698

**Pull request closed without merging**